core/ir/aarch64/codec.c

```diff
@@ -2385,49 +2372,44 @@ encode_opnd_x16immvs(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_
 static inline bool
 decode_opnd_vindex_H(uint enc, int opcode, byte *pc, OUT opnd_t *opnd)
 {
-    uint bits = (enc >> 11 & 1) << 2 | (enc >> 21 & 1) << 1 | (enc >> 20 & 1);
-    *opnd = opnd_create_immed_int(bits, OPSZ_2b);
+    /* Example encoding:
+     * FMLA <Vd>.<T>, <Vn>.<T>, <Vm>.H[<index>]
+     * 3322222222221111111111
```

Comment:
What is this 33222...?
Oh, is it the tens digit for the offsets below?  Hmm, ok.  Slightly confusing, along with the `z` on its own line (I figured out that it's part of `sz`) -- but not sure I have a much better idea: 2 digits for each perhaps but prob should just leave it.

Reply:
This was an attempt at commenting the `H:L:M` bit-twiddling for clarity and future maintenance.

I've used the same bit numbering scheme in `encode_opnd_fpimm8()`, `decode_opnd_fpimm13()` and other places but that divides the encoding into nibbles which is more obvious as a bit numbering comment I guess:
https://github.com/DynamoRIO/dynamorio/blob/d998d2bf371a53fe8c196c1152c75b5fa0c356ee/core/ir/aarch64/codec.c#L1821
https://github.com/DynamoRIO/dynamorio/blob/d998d2bf371a53fe8c196c1152c75b5fa0c356ee/core/ir/aarch64/codec.c#L2594

I can't think of a better way of commenting this sort of information in `ASCII` text.